packages/@sanity/desk-tool/src/panes/document/DocumentPane.tsx

```diff
@@ -67,7 +68,8 @@ export const DocumentPane = memo(function DocumentPane(props: DocumentPaneProvid
   const [templatePermissions, isTemplatePermissionsLoading] = useUnstableTemplatePermissions(
     getNewDocumentOptions()
   )
-  const isLoaded = isDocumentLoaded && !isTemplatePermissionsLoading
+  const [isReferenced, isReferenceLoading] = useDocumentsIsReferenced(options.id)
```

Comment:
Seeing the amount of places this needs to be ["prop-drilled"](https://blog.logrocket.com/solving-prop-drilling-react-apps), I am wondering if perhaps it would make sense to lift this up to the [DocumentPaneProvider](https://github.com/sanity-io/sanity/blob/next/packages/@sanity/desk-tool/src/panes/document/DocumentPaneProvider.tsx) and attach it to the [DocumentPaneContextValue](https://github.com/sanity-io/sanity/blob/next/packages/@sanity/desk-tool/src/panes/document/DocumentPaneContext.ts#L13). That way, you could utilize the `useDocumentPane()` hook in order to get the information instead of relying on the prop being passed down everywhere you might need it.

Reply:
This makes SO much sense - thanks for pointing this out 🙌